lib/ansible/modules/storage/netapp/na_ontap_interface.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/python
+""" this is interface module
+
+ (c) 2018, NetApp, Inc
+ # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+
+module: na_ontap_interface
+short_description: ONTAP LIF configuration
+
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: chhaya gunawat (chhayag@netapp.com)
+
+description:
+    - Creating / deleting and modifying the LIF.
+
+options:
+  state:
+    description:
+    - Whether the specified interface should exist or not.
+    choices: ['present', 'absent']
+    default: present
+
+  interface_name:
+    description:
+    - Specifies the logical interface (LIF) name.
+    required: true
+
+  home_node:
+    description:
+    - Specifies the LIF's home node.
+    - Required when C(state=present).
+
+  home_port:
+    description:
+    - Specifies the LIF's home port.
+    - Required when C(state=present)
+
+  role:
+    description:
```

Comment:
It makes sense to add the choices, but I'm concerned that the list can change.  ONTAP may deprecate some choices, or add new choices in a new release.
Here we are only the middle man.  

So if we list all choices, we are helping the end users, but this may become obsolete.
If we don't list the choices, we are to refer the end users to ONTAP documentation.

What's better?

Reply:
If you can't know valid values to pass the module without consulting external docs that may or may not be accessible to the person developing the content (as IIRC all the useful NetApp docs are behind a paywall that not everyone in the org may have access to), that kinda sucks.